validations/src/validations.rs

```diff
@@ -1677,6 +1757,65 @@ pub fn validate_block_transactions(
         );
     }
 
+    // validate stake transactions in a block
+    let mut st_mt = ProgressiveMerkleTree::sha256();
+    let mut st_weight: u32 = 0;
+
+    // Check if the block contains more than one stake tx from the same operator
+    let duplicate = block
+        .txns
+        .stake_txns
+        .clone()
+        .into_iter()
```

Comment:
Why cloning before iterating? I think you should be able to take a cloned iterator with `.iter().cloned()`, so that at most 1 item gets cloned.

Reply:
I don't even need to clone it if I use the referece in the map below